### ISSUE TITLE
Plugin fails repeatedly in Bulb.hasKelvin

### ISSUE THREAD

**Issue by smallstepstoday** [calvarium/homebridge-lifx-plugin#27]
**Describe The Bug:**
Plugin fails [here](https://github.com/calvarium/homebridge-lifx-plugin/blob/6f183d0210f65f6fc8305bac02a6678eb9b6c5d3/src/bulb.ts#L64):

Instead of:
```typescript
    return this.HardwareInfo?.productFeatures.temperature_range?.reduce((a, b) => b - a) || 0 > 0;
```
it should most likely read:
```typescript
    return this.HardwareInfo?.productFeatures?.temperature_range?.reduce((a, b) => b - a) || 0 > 0;
                                             ^
```

**To Reproduce:**
1. Install homebridge
2. Install homebridge-lifx-plugin
3. View logs

**Expected behavior:**
homebridge-lifx-plugin operates without error

**Logs:**

```
[2/4/2023, 7:56:38 PM] [HB Supervisor] Restarting Homebridge...
[2/4/2023, 7:56:38 PM] [HB Supervisor] Starting Homebridge with extra flags: -I
[2/4/2023, 7:56:39 PM] [HB Supervisor] Started Homebridge v1.6.0 with PID: 47670
[2/4/2023, 7:56:39 PM] Loaded config.json with 0 accessories and 2 platforms.
[2/4/2023, 7:56:39 PM] Loaded 7 cached accessories from cachedAccessories.
[2/4/2023, 7:56:39 PM] ---
[2/4/2023, 7:56:39 PM] Loaded plugin: homebridge-config-ui-x@4.50.2
[2/4/2023, 7:56:39 PM] Registering platform 'homebridge-config-ui-x.config'
[2/4/2023, 7:56:39 PM] ---
[2/4/2023, 7:56:39 PM] Loaded plugin: homebridge-lifx-plugin@0.2.4
[2/4/2023, 7:56:39 PM] Registering platform 'homebridge-lifx-plugin.LifxPlugin'
[2/4/2023, 7:56:39 PM] ---
[2/4/2023, 7:56:39 PM] Loading 2 platforms...
[2/4/2023, 7:56:39 PM] [Lifx] Initializing LifxPlugin platform...
[2/4/2023, 7:56:39 PM] [homebridge-lifx-plugin] This plugin generated a warning from the characteristic 'Firmware Revision': Characteristic not in required or optional characteristic section for service AccessoryInformation. Adding anyway.. See https://homebridge.io/w/JtMGR for more info.
[2/4/2023, 7:56:39 PM] [homebridge-lifx-plugin] This plugin generated a warning from the characteristic 'Firmware Revision': Characteristic not in required or optional characteristic section for service AccessoryInformation. Adding anyway.. See https://homebridge.io/w/JtMGR for more info.
[2/4/2023, 7:56:39 PM] [homebridge-lifx-plugin] This plugin generated a warning from the characteristic 'Firmware Revision': Characteristic not in required or optional characteristic section for service AccessoryInformation. Adding anyway.. See https://homebridge.io/w/JtMGR for more info.
[2/4/2023, 7:56:39 PM] [homebridge-lifx-plugin] This plugin generated a warning from the characteristic 'Firmware Revision': Characteristic not in required or optional characteristic section for service AccessoryInformation. Adding anyway.. See https://homebridge.io/w/JtMGR for more info.
[2/4/2023, 7:56:39 PM] [homebridge-lifx-plugin] This plugin generated a warning from the characteristic 'Firmware Revision': Characteristic not in required or optional characteristic section for service AccessoryInformation. Adding anyway.. See https://homebridge.io/w/JtMGR for more info.
[2/4/2023, 7:56:39 PM] [homebridge-lifx-plugin] This plugin generated a warning from the characteristic 'Firmware Revision': Characteristic not in required or optional characteristic section for service AccessoryInformation. Adding anyway.. See https://homebridge.io/w/JtMGR for more info.
[2/4/2023, 7:56:39 PM] [homebridge-lifx-plugin] This plugin generated a warning from the characteristic 'Firmware Revision': Characteristic not in required or optional characteristic section for service AccessoryInformation. Adding anyway.. See https://homebridge.io/w/JtMGR for more info.
Setup Payload:
...
Enter this code with your HomeKit app on your iOS device to pair with Homebridge:
                       
    ┌────────────┐     
     │ ...                                             │     
    └────────────┘     
                       
[2/4/2023, 7:56:39 PM] Homebridge v1.6.0 (HAP v0.11.0) (Homebridge 17E7) is running on port 51438.
[2/4/2023, 7:56:40 PM] [homebridge-lifx-plugin] This plugin generated a warning from the characteristic 'Manufacturer': characteristic value expected string and received undefined. See https://homebridge.io/w/JtMGR for more info.
[2/4/2023, 7:56:40 PM] [homebridge-lifx-plugin] This plugin generated a warning from the characteristic 'Model': characteristic value expected string and received undefined. See https://homebridge.io/w/JtMGR for more info.
[2/4/2023, 7:56:40 PM] TypeError: Cannot read properties of undefined (reading 'temperature_range')
    at Bulb.hasKelvin (.../.nvm/versions/node/v16.18.1/lib/node_modules/homebridge-lifx-plugin/src/bulb.ts:64:47)
    at LifxPlatformAccessory.bindFunctions (.../.nvm/versions/node/v16.18.1/lib/node_modules/homebridge-lifx-plugin/src/platformAccessory.ts:56:19)
    at .../.nvm/versions/node/v16.18.1/lib/node_modules/homebridge-lifx-plugin/src/platformAccessory.ts:25:12
    at .../.nvm/versions/node/v16.18.1/lib/node_modules/homebridge-lifx-plugin/src/bulb.ts:29:11
    at .../.nvm/versions/node/v16.18.1/lib/node_modules/homebridge-lifx-plugin/src/bulb.ts:90:7
    at .../.nvm/versions/node/v16.18.1/lib/node_modules/homebridge-lifx-plugin/src/bulb.ts:125:7
    at Client.<anonymous> (.../.nvm/versions/node/v16.18.1/lib/node_modules/homebridge-lifx-plugin/node_modules/lifx-lan-client/lib/lifx/light.js:236:5)
    at Client.<anonymous> (.../.nvm/versions/node/v16.18.1/lib/node_modules/homebridge-lifx-plugin/node_modules/lifx-lan-client/lib/lifx/client.js:451:26)
    at Array.forEach (<anonymous>)
    at Client.processMessageHandlers (.../.nvm/versions/node/v16.18.1/lib/node_modules/homebridge-lifx-plugin/node_modules/lifx-lan-client/lib/lifx/client.js:438:24)
[2/4/2023, 7:56:40 PM] Got SIGTERM, shutting down Homebridge...
[2/4/2023, 7:56:45 PM] [HB Supervisor] Homebridge Process Ended. Code: 143, Signal: null
[2/4/2023, 7:56:50 PM] [HB Supervisor] Restarting Homebridge...
[2/4/2023, 7:56:50 PM] [HB Supervisor] Starting Homebridge with extra flags: -I
[2/4/2023, 7:56:50 PM] [HB Supervisor] Started Homebridge v1.6.0 with PID: 47782
[2/4/2023, 7:56:50 PM] Loaded config.json with 0 accessories and 2 platforms.
[2/4/2023, 7:56:50 PM] Loaded 7 cached accessories from cachedAccessories.
[2/4/2023, 7:56:50 PM] ---
[2/4/2023, 7:56:50 PM] Loaded plugin: homebridge-config-ui-x@4.50.2
[2/4/2023, 7:56:50 PM] Registering platform 'homebridge-config-ui-x.config'
[2/4/2023, 7:56:50 PM] ---
[2/4/2023, 7:56:50 PM] Loaded plugin: homebridge-lifx-plugin@0.2.4
[2/4/2023, 7:56:50 PM] Registering platform 'homebridge-lifx-plugin.LifxPlugin'
[2/4/2023, 7:56:50 PM] ---
[2/4/2023, 7:56:50 PM] Loading 2 platforms...
[2/4/2023, 7:56:50 PM] [Lifx] Initializing LifxPlugin platform...
[2/4/2023, 7:56:50 PM] [homebridge-lifx-plugin] This plugin generated a warning from the characteristic 'Firmware Revision': Characteristic not in required or optional characteristic section for service AccessoryInformation. Adding anyway.. See https://homebridge.io/w/JtMGR for more info.
[2/4/2023, 7:56:50 PM] [homebridge-lifx-plugin] This plugin generated a warning from the characteristic 'Firmware Revision': Characteristic not in required or optional characteristic section for service AccessoryInformation. Adding anyway.. See https://homebridge.io/w/JtMGR for more info.
[2/4/2023, 7:56:50 PM] [homebridge-lifx-plugin] This plugin generated a warning from the characteristic 'Firmware Revision': Characteristic not in required or optional characteristic section for service AccessoryInformation. Adding anyway.. See https://homebridge.io/w/JtMGR for more info.
[2/4/2023, 7:56:50 PM] [homebridge-lifx-plugin] This plugin generated a warning from the characteristic 'Firmware Revision': Characteristic not in required or optional characteristic section for service AccessoryInformation. Adding anyway.. See https://homebridge.io/w/JtMGR for more info.
[2/4/2023, 7:56:50 PM] [homebridge-lifx-plugin] This plugin generated a warning from the characteristic 'Firmware Revision': Characteristic not in required or optional characteristic section for service AccessoryInformation. Adding anyway.. See https://homebridge.io/w/JtMGR for more info.
[2/4/2023, 7:56:50 PM] [homebridge-lifx-plugin] This plugin generated a warning from the characteristic 'Firmware Revision': Characteristic not in required or optional characteristic section for service AccessoryInformation. Adding anyway.. See https://homebridge.io/w/JtMGR for more info.
[2/4/2023, 7:56:50 PM] [homebridge-lifx-plugin] This plugin generated a warning from the characteristic 'Firmware Revision': Characteristic not in required or optional characteristic section for service AccessoryInformation. Adding anyway.. See https://homebridge.io/w/JtMGR for more info.
Setup Payload:
...
Enter this code with your HomeKit app on your iOS device to pair with Homebridge:
                       
    ┌────────────┐     
     │ ...                                             │     
    └────────────┘     
                       
[2/4/2023, 7:56:50 PM] Homebridge v1.6.0 (HAP v0.11.0) (Homebridge 17E7) is running on port 51438.
[2/4/2023, 7:56:51 PM] [homebridge-lifx-plugin] This plugin generated a warning from the characteristic 'Manufacturer': characteristic value expected string and received undefined. See https://homebridge.io/w/JtMGR for more info.
[2/4/2023, 7:56:51 PM] [homebridge-lifx-plugin] This plugin generated a warning from the characteristic 'Model': characteristic value expected string and received undefined. See https://homebridge.io/w/JtMGR for more info.
[2/4/2023, 7:56:51 PM] TypeError: Cannot read properties of undefined (reading 'temperature_range')
    at Bulb.hasKelvin (.../.nvm/versions/node/v16.18.1/lib/node_modules/homebridge-lifx-plugin/src/bulb.ts:64:47)

   ... repeats continuously ...
```

**Plugin Config:**

```json
{
   "name": "Lifx",
   "duration": 0,
   "brightnessDuration": 300,
   "colorDuration": 300,
   "broadcast": "255.255.255.255",
   "lightOfflineTolerance": 3,
   "messageHandlerTimeout": 45000,
   "resendPacketDelay": 150,
   "resendMaxTimes": 3,
   "debug": false,
   "autoDiscover": true,
   "platform": "LifxPlugin"
}
```

**Screenshots:**
N/A

**Environment:**

* **Plugin Version**: v0.2.4
* **Homebridge Version**: v1.6.0
* **Node.js Version**: v16.18.1
* **NPM Version**: 8.19.2
* **Operating System**: macOS Ventura

**Comment by lox** [comment]
Fixed in #29 and on my fork @smallstepstoday: https://github.com/lox/homebridge-lifx-plugin

**Comment by gagegreg** [comment]
I sat down tonight to fix this bug as it has been driving me crazy for the past few months.  My lifx is always restarting and the log says this each time.  Thanks for taking the time to fix it. @lox.  I will use your fork until its in the main. Cheers!
